### ISSUE TITLE
ipfshttpclient.exceptions.StatusError in ipfshttpclient 0.4.13.2 and go-ipfs v0.5.1

### ISSUE THREAD

**Issue by raminqaf** [ipfs-shipyard/py-ipfs-http-client#215]
Hi there, 
I updated my go-ipfs library to `v0.5.1` and upgraded the client form `0.4.12` to the latest version, which currently is `0.4.13.2`. After running my application I ran into this error message:
```
ipfshttpclient.exceptions.StatusError: HTTPError: 405 Client Error: Method Not Allowed for url: http://localhost:5001/api/v0/version?stream-channels=true
```
This error was not happening when I was using the `0.4.12` version of the client.
any ideas why this is happening?

**Comment by tabishimran** [comment]
I have the same issue! Can't figure it out, I'm facing it while using the docker container from https://hub.docker.com/r/ipfs/go-ipfs/

**Comment by Vshows** [comment]
I have the same issue!

**Comment by tabishimran** [comment]
https://github.com/ipfs-shipyard/py-ipfs-http-client/blob/5e3edf5f8e9e117738c77b5773c78d7b48942bc2/ipfshttpclient/http.py#L286

The error is being caused by this if statement which changes the 'POST' request to a 'HEAD' request, which is apparently not allowed.

It's a check for daemon workarounds, the daemon workaround gets added at 
https://github.com/ipfs-shipyard/py-ipfs-http-client/blob/ea0acc023a4cc0ceedb34ab76f56fccad5ede2f2/ipfshttpclient/client/__init__.py#L108

The code checks if the version of the IPFS daemon it connects to is less than 0.5, if it is, it adds that workaround which messes up things.

**Comment by alexander255** [comment]
Please try with the master from the git repository, your issue is likely not present there. There is an upcoming 0.5.0 release pending, but not released yet which will have full compatibility with this version, as well as a few minor breaking changes/cleanups.

**Comment by SaltyLeo** [comment]
I have the same issue!

**Comment by Twodragon0** [comment]
I have the same issue. It's working fine before change v0.5.1. After changed code, have this error. 
AttributeError: module 'ipfshttpclient.utils' has no attribute 'deep_update'

**Comment by cai-harry** [comment]
Same issue here!

**Comment by fubuloubu** [comment]
@alexander255 Is there anyway I can help on the 0.5.0 release that is pending? I really need this release, as there is a dependency that uses this library, so it's not as easy for me to pin to master as a workaround

**Comment by mikeshultz** [comment]
@alexander255  I can confirm the initial reported error has been resolved in master.

**Comment by ntninja** [comment]
Fixed in 0.6.0
